### PR TITLE
Add a tidy check to find unexpected files in UI tests, and clean up the results

### PR DESCRIPTION
**Pull request by asquared31415**
While looking at UI tests, I noticed several weird files that were not being tested, some from even pre-1.0. I added a tidy check that fails if any files not known to compiletest or not used in tests (via manual list) are present in the ui tests.

Unfortunately the root entry limit had to be raised by 1 to accommodate the stderr file for one of the tests.

r? @fee1-dead

### CODE REVIEW

File: src/tools/tidy/src/ui_tests.rs [hunk 1]
```diff
@@ -66,7 +93,19 @@ pub fn check(path: &Path, bad: &mut bool) {
...
+                // note: checking that "not any" of the paths match is checking if none match
+                if !EXTENSION_EXCEPTION_PATHS.iter().any(|path| file_path.ends_with(path)) {
+                    tidy_error!(
+                        bad,
+                        "file {} has unexpected extension {}",
+                        file_path.display(),
+                        ext
+                    );
+                }
+            }
```
Comment (WaffleLapkin) [inline src/tools/tidy/src/ui_tests.rs:108]:
```suggestion
            // check for files that don't have an expected extension, they're probably typos or not meant to be here
            if !EXPECTED_TEST_FILE_EXTENSIONS.contains(&ext)
                // note: checking that "not any" of the paths match is checking if none match
                && !EXTENSION_EXCEPTION_PATHS.iter().any(|path| file_path.ends_with(path))
            {
                tidy_error!(
                    bad,
                    "file {} has unexpected extension {ext}",
                    file_path.display(),
                );
            }
```

Comment (WaffleLapkin) [reply src/tools/tidy/src/ui_tests.rs:108]: Btw alternatively the second check can be written as `.all(|| !ends_with)` which is probably more intuitive

File: src/tools/tidy/src/ui_tests.rs [hunk 2]
```diff
@@ -4,13 +4,40 @@
...
+
+const EXPECTED_TEST_FILE_EXTENSIONS: &[&str] = [
+    "rs",     // test source files
+    "stderr", // expected stderr file, corresponds to a rs file
+    "stdout", // expected stdout file, corresponds to a rs file
+    "fixed",  // expected source file after applying fixes
+    "md",     // test directory descriptions
+    "ftl",    // translation tests
+]
+.as_slice();
```
Comment (klensy) [inline src/tools/tidy/src/ui_tests.rs:24]: = &["rs", ...], no need to as_slice()

File: src/tools/tidy/src/ui_tests.rs [hunk 3]
```diff
@@ -4,13 +4,40 @@
...
+    "tests/ui/crate-loading/auxiliary/libfoo.rlib", // testing loading a manually created rlib
+    "tests/ui/include-macros/data.bin", // testing including data with the include macros
+    "tests/ui/include-macros/file.txt", // testing including data with the include macros
+    "tests/ui/macros/macro-expanded-include/file.txt", // testing including data with the include macros
+    "tests/ui/macros/not-utf8.bin", // testing including data with the include macros
+    "tests/ui/macros/syntax-extension-source-utils-files/includeme.fragment", // more include
+    "tests/ui/unused-crate-deps/test.mk", // why would you use make
+    "tests/ui/proc-macro/auxiliary/included-file.txt", // more include
+]
+.as_slice();
```
Comment (fee1-dead) [inline src/tools/tidy/src/ui_tests.rs:40]: also this

**Review by fee1-dead — APPROVED** [review]
r=me with nits above addressed

File: src/bootstrap/Cargo.lock [hunk 4]
```diff
@@ -99,9 +99,9 @@ checksum = "baf1de4339761588bc0619e3cbc0120ee582ebb74b53b4efbf79117bd2da40fd"
 
 [[package]]
 name = "clap"
-version = "4.2.7"
+version = "4.2.4"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "34d21f9bf1b425d2968943631ec91202fe5e837264063503708b83013f8fc938"
+checksum = "956ac1f6381d8d82ab4684768f89c0ea3afe66925ceadb4eeb3fc452ffc55d62"
```
Comment (klensy) [inline src/bootstrap/Cargo.lock]: And squash, as some comments touches cargo.lock in between.

**Pull request merged**